File: api/Bookshop.Application/Features/Categories/CategoryResponseDto.cs
using Bookshop.Application.Features.Dto;

namespace Bookshop.Application.Features.Categories
{
    public record CategoryResponseDto
    {
        public long Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public bool IsVisible { get; set; }
    }
}

File: ui/src/app/dto/category/category-response-dto.ts
export class CategoryResponseDto {
  id: number = 0;
  title: string = '';
  description: string = '';
  isVisible: boolean = true;

  constructor(data?: Partial<CategoryResponseDto>) {
    this.id = data?.id ?? this.id;
    this.title = data?.title ?? this.title;
    this.description = data?.description ?? this.description;
    this.isVisible = data?.isVisible ?? this.isVisible;
  }
}